libs/langchain/langchain/vectorstores/chroma.py

```diff
@@ -92,6 +92,13 @@ def __init__(
             self._persist_directory = persist_directory
         else:
             if client_settings:
+                # Check the client settings required input values
+                if client_settings.chroma_db_impl is None or \
+                    client_settings.chroma_db_impl == "":
+                    raise ValueError("client_settings.chroma_db_impl is required")
+                if client_settings.persist_directory is None or \
+                    client_settings.persist_directory == "":
+                    raise ValueError("client_settings.persist_directory is required")
```

Comment:
is this actually required?

Reply:
No, it's not. Chroma has 3 modes

`chromadb.EphemeralClient()` - runs in memory
`chromadb.PersistentClient()` - runs in memory and saves to disk
`chromadb.HttpClient()` - talks to a remote chroma server